src/lib/finishPollMessage.ts

```diff
@@ -26,33 +43,66 @@ export async function finishPollMessage({ data, read, persistence, modify }: {
     const poll = await getPoll(String(data.message.id), read);
 
     if (!poll) {
-        throw new Error('no such poll');
+        throw new Error("no such poll");
     }
 
     if (poll.finished) {
-        throw new Error('this poll is already finished');
+        throw new Error("this poll is already finished");
     }
 
     if (poll.uid !== data.user.id) {
-        throw new Error('You are not allowed to finish the poll'); // send an ephemeral message
+        throw new Error("You are not allowed to finish the poll"); // send an ephemeral message
     }
 
     try {
         await finishPoll(poll, { persis: persistence });
 
-        const message = await modify.getUpdater().message(data.message.id as string, data.user);
+        const message = await modify
+            .getUpdater()
+            .message(data.message.id as string, data.user);
         message.setEditor(message.getSender());
 
         const block = modify.getCreator().getBlockBuilder();
 
-        const showNames = await read.getEnvironmentReader().getSettings().getById('use-user-name');
-
-        createPollBlocks(block, poll.question, poll.options, poll, showNames.value);
+        const showNames = await read
+            .getEnvironmentReader()
+            .getSettings()
+            .getById("use-user-name");
+        const useInternet = await read
+            .getEnvironmentReader()
+            .getSettings()
+            .getById("use-public-internet");
+        createPollBlocks(
+            block,
+            poll.question,
+            poll.options,
+            poll,
+            showNames.value,
+            useInternet.value
+        );
 
         message.setBlocks(block);
 
-        return modify.getUpdater().finish(message);
+        modify.getUpdater().finish(message);
+
+        if (poll.wordcloud && useInternet.value) {
+            const attachment = <IMessageAttachment>{
+                imageUrl: `https://quickchart.io/wordcloud?text=${poll.options.join(
```

Comment:
2 points here
1. If we're hard-coding this url here then how would the organisations who have hosted their own quickcharts server connect this app to it? For this reason, I'd earlier suggested to keep this api url as a app setting.

2. Are we planning to support paid plans for quickcharts? On the pricing page of quickcharts, it says that it has some rate limitation for the free plan - if that's the case then this might be an issue with bigger communities and for such big communities we'd need to add support for their paid plans.

Reply:
Thanks Murtaza.
1. I realized (on contacting Ian from Quickchart) that they haven't integrated the word-cloud API in their [open-source](https://github.com/typpo/quickchart) codebase so there wouldn't be much use hosting a Quickchart server. The only way to access the word-cloud API is through the public URL which some organizations might not have access to. Hence the `boolean` app setting determining whether the org has access or not.
2. I'm unsure whether word-cloud hits would be included within the paid version since it hasn't been integrated. Moreover, the current rate-limitation is [1 chart/sec](https://quickchart.io/qr-code-api/) which I think should be pretty scalable.

Let me know your thoughts.

Reply:
1. If that's the case, then I think we don't have any option here with QuickChart :disappointed: Did he happen to mention when they will be allowing this fetaure? Also during your research did you find any alternative chart generator which provides on-premise hosting? I'm asking this because if we're introducing this public internet access limitation in this app, then the communities using Rocket.Chat within on-premise would miss out on this feature.
2. They say its a `60 requests per minute per IP rate limit` so if all the users are accessing the room from a different network then the limit is fair enough, however if all the users are on a private network behind a (Router + NAT) then although each of them will have a different local IP, but the same public IP - In that case if there is a single room with more than 60 users, they all would be identified as a single entity/IP and the rate limit will come into picture. Please give some thought on this and check if we can add support for the paid plan. Thanks.

Reply:
1. Unfortunately no. They seem to be in a fix whether to include it within the docker image or spin it out as a separate image. They don't seem to have time-boxed this change.
I found a few on RapidAPI but they don't seem to have a locally hosted or open-source version and are all paid.

2.  Valid point. I've mailed inquiring whether word-cloud is included within the paid plan. Will update here. Thanks.